### ISSUE TITLE
I3042 wrongly identifies API paths as ARNs

### ISSUE THREAD

**Issue by ohadbenita** [aws-cloudformation/cfn-python-lint#1976]
*cfn-lint version: 0.48.2*

*Description of issue.* API paths are considered ARNs and are mistakenly identified by [I3042](https://github.com/aws-cloudformation/cfn-python-lint/blob/master/docs/rules.md#I3042)

A sample failing template (SAM):
```yaml
Transform: AWS::Serverless-2016-10-31
Globals:
  Function:
    Runtime: provided
    Handler: index.handler
Resources:
  MyApi:
    Type: AWS::Serverless::Api
    Properties:
      Name: my-api
      StageName: prod
      MethodSettings:
        - LoggingLevel: INFO
          MetricsEnabled: true
          ResourcePath: "/*"
          HttpMethod: "*"
      Cors:
        AllowMethods: "'*'"
        AllowHeaders: "'*'"
        AllowOrigin: "'*'"

  ExampleLambdaFunction:
    Type: AWS::Serverless::Function
    Properties:
      FunctionName: example-function-name
      Handler: com.acme.lambda.function
      Role: !GetAtt ExampleLambdaFunctionRole.Arn
      Events:
        ApiEvent:
          Type: Api
          Properties:
            Method: GET
            Path: /mypath
            RestApiId: !Ref MyApi
```

**Comment by PatMyron** [comment]
So it looks like transformed templates are actually hardcoding partitions in ARNs, +1 https://github.com/aws/serverless-application-model/issues/1590
The mapping from the violation in the transformed template back to the line in the original source template is a bit best effort

```bash
$ cfn-lint --info t.yaml 
2021-04-11 17:31:06,775 - cfnlint - INFO - SAM Translator: 1.35.0
2021-04-11 17:31:06,775 - cfnlint - INFO - Setting AWS_DEFAULT_REGION to us-east-1
2021-04-11 17:31:06,859 - cfnlint - INFO - Transformed template:
```
<details>
  <summary>Transformed template:</summary>
  
```yaml
{
  "Resources": {
    "ExampleLambdaFunction": {
      "Properties": {
        "Code": {
          "S3Bucket": "bucket",
          "S3Key": "value"
        },
        "FunctionName": "example-function-name",
        "Handler": "com.acme.lambda.function",
        "Role": {
          "Fn::GetAtt": [
            "ExampleLambdaFunctionRole",
            "Arn"
          ]
        },
        "Runtime": "provided",
        "Tags": [
          {
            "Key": "lambda:createdBy",
            "Value": "SAM"
          }
        ]
      },
      "Type": "AWS::Lambda::Function"
    },
    "ExampleLambdaFunctionApiEventPermissionprod": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Ref": "ExampleLambdaFunction"
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Sub": [
            "arn:aws:execute-api:${AWS::Region}:${AWS::AccountId}:${__ApiId__}/${__Stage__}/GET/mypath",
            {
              "__ApiId__": {
                "Ref": "MyApi"
              },
              "__Stage__": "*"
            }
          ]
        }
      },
      "Type": "AWS::Lambda::Permission"
    },
    "MyApi": {
      "Properties": {
        "Body": {
          "info": {
            "title": {
              "Ref": "AWS::StackName"
            },
            "version": "1.0"
          },
          "paths": {
            "/mypath": {
              "get": {
                "responses": {},
                "x-amazon-apigateway-integration": {
                  "httpMethod": "POST",
                  "type": "aws_proxy",
                  "uri": {
                    "Fn::Sub": "arn:aws:apigateway:${AWS::Region}:lambda:path/2015-03-31/functions/${ExampleLambdaFunction.Arn}/invocations"
                  }
                }
              },
              "options": {
                "consumes": [
                  "application/json"
                ],
                "produces": [
                  "application/json"
                ],
                "responses": {
                  "200": {
                    "description": "Default response for CORS method",
                    "headers": {
                      "Access-Control-Allow-Headers": {
                        "type": "string"
                      },
                      "Access-Control-Allow-Methods": {
                        "type": "string"
                      },
                      "Access-Control-Allow-Origin": {
                        "type": "string"
                      }
                    }
                  }
                },
                "summary": "CORS support",
                "x-amazon-apigateway-integration": {
                  "requestTemplates": {
                    "application/json": "{\n  \"statusCode\" : 200\n}\n"
                  },
                  "responses": {
                    "default": {
                      "responseParameters": {
                        "method.response.header.Access-Control-Allow-Headers": "'*'",
                        "method.response.header.Access-Control-Allow-Methods": "'*'",
                        "method.response.header.Access-Control-Allow-Origin": "'*'"
                      },
                      "responseTemplates": {
                        "application/json": "{}\n"
                      },
                      "statusCode": "200"
                    }
                  },
                  "type": "mock"
                }
              }
            }
          },
          "swagger": "2.0"
        },
        "Name": "my-api"
      },
      "Type": "AWS::ApiGateway::RestApi"
    },
    "MyApiDeployment3a564585a0": {
      "Properties": {
        "Description": "RestApi deployment id: 3a564585a08350704b5b9894c62bb7e8de6fafe1",
        "RestApiId": {
          "Ref": "MyApi"
        },
        "StageName": "Stage"
      },
      "Type": "AWS::ApiGateway::Deployment"
    },
    "MyApiprodStage": {
      "Properties": {
        "DeploymentId": {
          "Ref": "MyApiDeployment3a564585a0"
        },
        "MethodSettings": [
          {
            "HttpMethod": "*",
            "LoggingLevel": "INFO",
            "MetricsEnabled": true,
            "ResourcePath": "/*"
          }
        ],
        "RestApiId": {
          "Ref": "MyApi"
        },
        "StageName": "prod"
      },
      "Type": "AWS::ApiGateway::Stage"
    }
  }
}
```
</details>
